### DISCUSSION TITLE
is it possible to filter text by language?

### DISCUSSION THREAD

**Discussion by austin23cook** [nianeyna/ao3downloader#58] [Q&A] [answered]
im downloading a big chunk of scrapes for training a softprompt on a language generation AI and i need quite a large sample size but the amount of data is a bit too much to go through with anything but regex once its all done (im expecting quite a few GB of raw text files. so even scrolling through as fast as i can could take hours)

**Comment by nianeyna** [comment, marked as answer]
You can use any of ao3's native searching and filtering tools with ao3downloader - just apply filters on the site as you normally would, and then copy the link from the address bar when you're done. This would mean doing a separate download for each language you want. Alternatively, if you don't want to do separate downloads, you could try importing the downloaded fics into Calibre and sorting through them there. I believe Calibre should be able to pick up the language information from the file, though if you download in HTML or PDF it might not work as well. Either way there's a risk (likelihood, even, if you're downloading a very large amount of works) that some works will be misfiled because the author selected the wrong language for whatever reason. Not much of a way around that that I'm aware of, just something to take into account.
